service/controller/resource/promxy/resource.go

```diff
@@ -99,7 +106,7 @@ func (r *Resource) toServerGroup(cluster metav1.Object) (*ServerGroup, error) {
 				Regex:     relabel.MustNewRegexp(fmt.Sprintf("prometheus-meta-operator;prometheus;%s", key.ClusterID(cluster))),
 			},
 		},
-		KubernetesSDConfigs: []*kubernetes.SDConfig{
+		Hosts: discovery.Configs{
```

Comment:
I could not make it serialize but good if it works :D

Reply:
That's my only worry here. This have to be tested.

Reply:
I had an issue where Config could not be serialized due to it being an interface. Hopefully, you will figure it out :D